### PR TITLE
Fix memory fences on x86

### PR DESCRIPTION
**Pull request by gacholio**
- dword rather than qword is sufficient for readWriteBarrier
- no memory fence is required for readBarrier
- remove lfence/sfence TODOs (not required for normal reads/writes)

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
Spoke with @0dvictor and confirmed that any locked instruction is sufficient for a memory barrier, and since the JIT and the Windows __faststorefence primitive also use dword, we may we as well line up.

LFENCE/SFENCE are only required for ordering of non-temporal SSE instructions.

File: include_core/AtomicSupport.hpp [hunk 1]
```diff
@@ -259,12 +259,7 @@ class VM_AtomicSupport
 #elif defined(_MSC_VER)
 		_ReadWriteBarrier();
 #elif defined(__GNUC__)
-#if defined(J9X86) || defined(J9HAMMER)
-		asm volatile("":::"memory");
-		/* TODO investigate whether or not we should call this
-		asm volatile("sfence" ::: "memory");
-		*/
-#elif defined(ARM) /* defined(J9X86) || defined(J9HAMMER) */
+#if defined(ARM)
```
Comment (DanHeidinga) [inline include_core/AtomicSupport.hpp]: line 268 needs an update `#endif /* defined(J9X86) || defined(J9HAMMER) */` to fix the comment

File: include_core/AtomicSupport.hpp [hunk 2]
```diff
@@ -292,24 +287,8 @@ class VM_AtomicSupport
...
-		/* TODO investigate whether or not we should call this instead
-		asm volatile("lfence":::"memory");
-		*/
-#elif defined(J9HAMMER)
-		asm volatile("lock orq $0x0,(%%rsp)" ::: "memory");
-		/* TODO investigate whether or not we should call this instead
-		asm volatile("lfence":::"memory");
-		*/
-#elif defined(ARM) /* defined(J9HAMMER) */
+#if defined(ARM)
```
Comment (DanHeidinga) [inline include_core/AtomicSupport.hpp]: The comment on line 297 needs to be updated as well

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by DanHeidinga** [comment]
> Did I read this correctly?

I had miss'd the last case was an `#else`, not another `#elif`

File: include_core/AtomicSupport.hpp (hunk 2 above)
Comment (gacholio) [reply include_core/AtomicSupport.hpp]: Comments addressed.

**Comment by gacholio** [comment]
Accidentally merged other changes - closing and starting again,

**Comment by gacholio** [comment]
New PR: https://github.com/eclipse/omr/pull/2572

**Pull request closed without merging**